core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -614,6 +585,74 @@ private[worker] class Worker(
     webUi.stop()
     metricsSystem.stop()
   }
+
+  private def trimFinishedExecutorsIfNecessary(): Unit = {
+    if (finishedExecutors.size > retainedExecutors) {
+      finishedExecutors.take(math.max(finishedExecutors.size / 10, 1)).foreach {
```

Comment:
Aren't these data structures also read by the web ui?  Do those reads go through the actor?